numpy/testing/_private/utils.py

```diff
@@ -2734,9 +2734,18 @@ def run_threaded(func, max_workers=8, pass_count=False,
                 barrier = threading.Barrier(max_workers)
                 args.append(barrier)
             if pass_count:
-                futures = [tpe.submit(func, i, *args) for i in range(max_workers)]
+                all_args = [(func, i, *args) for i in range(max_workers)]
             else:
-                futures = [tpe.submit(func, *args) for _ in range(max_workers)]
+                all_args = [(func, *args) for i in range(max_workers)]
+            try:
+                n_submitted = 0
+                futures = []
+                for arg in all_args:
+                    futures.append(tpe.submit(*arg))
+                    n_submitted += 1
+            finally:
+                if n_submitted < max_workers and pass_barrier:
+                    barrier.abort()
```

Comment:
I think this may be clearer to write as:
```
except:  # or BaseExecption if you prefer
    if pass_barrier:
        barrier.abort()
    raise
```
and avoid counting the `n_submitted` above.

Reply:
> (but maybe I missed a subtlety?)

bare `except:` and `except BaseException` is the sort of thing that linters mark as a code smell or a naive reader in the future would replace with `except Exception` (which is wrong). I like`try/finally` because it matches the intention and uses a language feature that linters won't flag.

I can get rid of `n_submitted` and just check `len(futures)` though.